app/uk/gov/hmrc/transitmovementsrouter/services/XmlParser.scala

```diff
@@ -40,14 +33,6 @@ object XmlParser extends XmlParsingServiceHelpers {
     }
     .single("referenceNumber")
 
-  def messageSenderWriter(messageType: MessageType, messageSender: MessageSender): Flow[ParseEvent, ParseEvent, NotUsed] = Flow[ParseEvent]
-    .mapConcat(
-      element =>
-        if (isElement(messageType.rootNode, element))
-          Seq(element, StartElement("messageSender"), Characters(s"${messageSender.value}"), EndElement("messageSender"))
-        else Seq(element)
-    )
-
   private def isElement(name: String, event: ParseEvent): Boolean = event match {
```

Comment:
One more thing, as you removed the code that used this, this is no-longer used, so this should be removed as well (it's good to make sure we don't keep things around we're not using, if we need it again, we can always get it out of the version control history).

Reply:
thank you, I understand I will delete it now